aviary/subsystems/propulsion/hamilton_standard.py

```diff
@@ -0,0 +1,1021 @@
+    prob.set_val(Dynamic.Mission.SHAFT_POWER, val=1850.0, units='hp')
+    prob.set_val(Dynamic.Mission.DENSITY, val=RHO_SEA_LEVEL_ENGLISH, units='slug/ft**3')
+    prob.set_val(Dynamic.Mission.VELOCITY, val=0.0, units='knot')
+    prob.set_val(Dynamic.Mission.TEMPERATURE, val=TSLS_DEGR, units='degR')
+    prob.run_model()
+
+    print(f"  density_ratio: {prob.get_val('prehs.density_ratio')}")
+    print(f"  adv_ratio: {prob.get_val('prehs.adv_ratio')}")
+    print(f"  power_coefficient: {prob.get_val('prehs.power_coefficient')}")
+
+    # 10.E10 is causing just a little numerical noise
+    data = prob.check_partials(
+        compact_print=True, show_only_incorrect=True, minimum_step=1e-12)
+
+
+if __name__ == "__main2__":
+    model = om.Group()
+    hs = model.add_subsystem('hs', HamiltonStandard(), promotes=['*'])
+    prob = om.Problem(model)
+    prob.setup()
+
+    prob.set_val('hs.power_coefficient', 0.2352124866909703)  # 0.235239178
+    prob.set_val('hs.adv_ratio', 0.0)
+    prob.set_val(Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR, 114)
+    prob.set_val(Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICENT, 0.5)
+    prob.set_val('hs.mach', 0.0)
+    prob.set_val('hs.tip_mach', 0.7154230531595535)  # 0.71542275
+    hs.options.set(num_blades=4)
+    print("Inputs to HamiltonStandard:")
+    print(f"  power_coefficient: {prob.get_val('hs.power_coefficient')}")
+    print(f"  adv_ratio: {prob.get_val('hs.adv_ratio')}")
+    print(f"  act_fac: {prob.get_val(Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR)}")
+    print(f"  cli: {prob.get_val(Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICENT)}")
+    print(f"  mach: {prob.get_val('hs.mach')}")
+    print(f"  tip_mach: {prob.get_val('hs.tip_mach')}")
+    print(f"  num_blades: {hs.options['num_blades']}")
+
+    prob.run_model()
+    print()
+    print("Outputs:")
+    print(f"  3/4 Blade Angle: {prob['hs.ang_blade'][0]}")
+    print(f"  CT: {prob['hs.thrust_coefficient'][0]}")
+    print(f"  XFT: {prob['hs.comp_tip_loss_factor'][0]}")
+
+
+if __name__ == "__main__":
+    model = om.Group()
+    prehs = model.add_subsystem('posths', PostHamiltonStandard(), promotes=['*'])
+    prob = om.Problem(model)
+    prob.set_solver_print(level=1)
+    # prob.setup(mode='rev')
+    prob.setup()
+
+    prob.set_val(Dynamic.Mission.PROPELLER_TIP_SPEED, 785.0, units='ft/s')
+    prob.set_val(Aircraft.Engine.PROPELLER_DIAMETER, 12.0, units='ft')
+    prob.set_val('power_coefficient', 0.19062805, units='unitless')
+    prob.set_val('adv_ratio', 0.0, units='unitless')
+    prob.set_val('density_ratio', 1.0, units='unitless')
+    prob.set_val('thrust_coefficient', 0.27650621496575234, units='unitless')
+    prob.set_val('comp_tip_loss_factor', 1.0, units='unitless')
+    prob.set_val(Dynamic.Mission.INSTALLATION_LOSS_FACTOR, 0.05, units='unitless')
+    prob.run_model()
+
+    data = prob.check_partials(
+        compact_print=True, show_only_incorrect=True, minimum_step=1e-12)
```

Comment:
should these be moved to tests?

Reply:
removed for now; Xun is working on partials tests, don't want that to hold up this PR